cli/src/commands/git/push.rs

```diff
@@ -173,6 +174,18 @@ pub struct GitPushArgs {
         add = ArgValueCompleter::new(complete::revset_expression_all),
     )]
     revisions: Vec<RevisionArg>,
+    /// Push all tags
+    #[arg(long)]
+    tags: bool,
```

Comment:
Since bookmarks and tags are modeled in a quite similar way in jj, I expect `jj git push` will by default push bookmarks and tags to the tracked remote if they are in the default revset range. `--tag=NAME` will behave in the same way as `--bookmark=NAME`.

If https://github.com/jj-vcs/jj/issues/8387 is accepted, newly-created tags can be pushed by `--tag=NAME`. I assume we'll also add `jj tag track`/`untrack` and `auto-track-tags` config.

Reply:
Thanks for giving some background.

That said, I don't fully understand what you're suggesting I should do or change. Apologies, I'm just getting started, and this is my first contribution to jj.

Are you suggesting that I wait until https://github.com/jj-vcs/jj/pull/8239 is accepted? Is there functionality that you think should be part of this changeset that current is not? Or are you just providing some additional context?

Reply:
In short, this isn't an easy task. We'll first need to think about how fetched Git tags will be mapped to `view.remote_views[remote].tags`. We'll do "force-with-lease" push by using `view.remote_views[remote].tags[name]` as the known state. I'm doing some related work, but I don't know when it will complete.

Regarding the UI, my point is that we don't need `--tags` option. `jj tag set foo && jj git push` should push the tag `foo` if `auto-track-tags` is configured and if the tag points to revisions to be pushed by default. `--tag=NAME` will behave exactly the same way as `--bookmark=NAME`.

Reply:
I see. In that case, would you recommend that I step back from this issue for the moment? Perhaps I should start with something easier, and loop back to this once I have a better understanding of the codebase and roadmap